src/Nri/Ui/Table/V8.elm

```diff
@@ -0,0 +1,368 @@
+    case cellType of
+        RowHeaderCell ->
+            th (Attributes.scope "row" :: attrs)
+
+        DataCell ->
+            td attrs
+
+
+{-| A column that renders some aspect of a value as text
+-}
+string :
+    { header : String
+    , value : data -> String
+    , width : LengthOrAuto compatible
+    , cellStyles : data -> List Style
+    , sort : Maybe SortDirection
+    }
+    -> Column data msg
+string { header, value, width, cellStyles, sort } =
+    Column (Html.text header) (value >> Html.text) (Css.width width) cellStyles sort DataCell False
+
+
+{-| Creates a placeholder column which will reserve the space for the column,
+this can be used when multiple tables have similar data and we want to keep
+the size consistent.
+-}
+placeholderColumn :
+    { width : LengthOrAuto compatible
+    }
+    -> Column data msg
+placeholderColumn options =
+    Column (Html.text "") (always (Html.text "")) (Css.width options.width) (always []) Nothing DataCell False
+
+
+{-| A column that renders however you want it to
+-}
+custom :
+    { header : Html msg
+    , view : data -> Html msg
+    , width : LengthOrAuto compatible
+    , cellStyles : data -> List Style
+    , sort : Maybe SortDirection
+    }
+    -> Column data msg
+custom options =
+    Column options.header options.view (Css.width options.width) options.cellStyles options.sort DataCell True
+
+
+{-| A column whose cells are row headers
+-}
+rowHeader :
+    { header : Html msg
+    , view : data -> Html msg
+    , width : LengthOrAuto compatible
+    , cellStyles : data -> List Style
+    , sort : Maybe SortDirection
+    }
+    -> Column data msg
+rowHeader options =
+    Column options.header options.view (Css.width options.width) options.cellStyles options.sort RowHeaderCell False
+
+
+
+-- VIEW
+
+
+{-| Displays a table of data without a header row
+-}
+viewWithoutHeader : { additionalStyles : List Style, alternatingRowColors : Bool } -> List (Column data msg) -> List data -> Html msg
+viewWithoutHeader { additionalStyles, alternatingRowColors } columns =
+    tableWithoutHeader additionalStyles columns (viewRow columns alternatingRowColors)
+
+
+{-| Displays a table of data based on the provided column definitions
+-}
+view : { additionalStyles : List Style, alternatingRowColors : Bool } -> List (Column data msg) -> List data -> Html msg
+view { additionalStyles, alternatingRowColors } columns =
+    tableWithHeader additionalStyles columns (viewRow columns alternatingRowColors)
+
+
+viewRow : List (Column data msg) -> Bool -> data -> Html msg
+viewRow columns alternatingRowColors data =
+    tr
+        [ css (rowStyles alternatingRowColors) ]
+        (List.map (viewColumn data) columns)
+
+
+viewColumn : data -> Column data msg -> Html msg
+viewColumn data (Column _ renderer width cellStyles _ cellType _) =
+    cell cellType
+        [ css ([ width, verticalAlign middle, padding4 (px 11) (px 12) (px 14) (px 12), textAlign left ] ++ cellStyles data)
```

Comment:
This adds top and bottom padding to internal cells. Are you sure we want that? My understanding is that we just wanted to align the start of headers and content cells:

Before:
<img width="1019" alt="Screenshot 2024-11-04 at 12 18 40 PM" src="https://github.com/user-attachments/assets/1858c8f3-43d2-4fcd-86e8-25b5cc55a64a">



After:
<img width="1021" alt="Screenshot 2024-11-04 at 12 18 44 PM" src="https://github.com/user-attachments/assets/3da4e313-2819-4e27-bc05-634e2bc660da">

Reply:
Hmm, the title of the ticket "consistent default padding for header and data cells", that is why I used the same padding including vertical padding, but I can remove the vertical padding.

Reply:
@staceyadams can probably confirm what the desired behavior is!